clearwater-infrastructure/usr/share/clearwater/bin/view-subscriber

```diff
@@ -0,0 +1,203 @@
+            else:
+                # The first public identity is the primary public identity. We
+                # save this off and also extract the IMS subscription XML, and
+                # all of the charging addresses.
+                primary_impu = reg_data.find('IMSSubscription/ServiceProfile/PublicIdentity/Identity').text
+
+                ims_subscription = ET.tostring(reg_data.find('IMSSubscription'))
+
+                ccf_addresses = [(ccf.text, ccf.attrib['priority']) for ccf in reg_data.find('ChargingAddresses').findall('CCF')]
+                ecf_addresses = [(ecf.text, ecf.attrib['priority']) for ecf in reg_data.find('ChargingAddresses').findall('ECF')]
+        except (ET.ParseError, AttributeError, KeyError):
+            sys.exit('Registration information XML returned in unexpected format.\n'
+                     'Please contact your system administrator.')
+    elif reg_data_r.status_code == 502:
+        sys.exit('Unable to contact HSS.\nPlease contact your system administrator.')
+    elif reg_data_r.status_code == 503:
+        sys.exit('HSS gateway is currently overloaded.\nPlease try again later.')
+    else:
+        sys.exit('Unable to retrieve subscriber\'s subscription XML due to unexpected '
+                 '{} error.\nPlease contact your system administrator.'.format(reg_data_r.status_code))
+
+    # Now we can build the Sprout API URLs (with the primary public identity).
+    bindings_url = 'http://{}/impu/{}/bindings'.format(sprout_api_address,
+                                                       primary_impu)
+    subscriptions_url = 'http://{}/impu/{}/subscriptions'.format(sprout_api_address,
+                                                                 primary_impu)
+
+    # If we fail to retrieve bindings or subscriptions we still display the
+    # information we've already retrieved. Therefore we build error strings.
+    bindings_err = ''
+    subscriptions_err = ''
+
+    try:
+        bindings_r = requests.get(bindings_url)
+
+        if bindings_r.status_code == 200:
+            try:
+                # Extract the URIs and expiry times for each binding.
+                bindings_json = bindings_r.json()
+                bindings = [(binding['uri'], binding['expires']) for (urn, binding) in bindings_json['bindings'].iteritems()]
+            except ValueError:
+                bindings_err = 'Bindings information is not valid JSON.\n' \
+                               '\tPlease contact your system administrator.'
+            except KeyError as e:
+                bindings_err = 'Bindings information returned in unexpected format ' \
+                               '- {} key not found in JSON.\n\tPlease contact your system administrator.'.format(e)
+        elif bindings_r.status_code == 404:
+            # This is not an error - the subscriber may just not be registered.
+            bindings_err = '(none)'
```

Comment:
bindings will not be defined if we enter this branch, and also, I'm quite keen to keep this branch (even if it were empty) because it reflects the API accurately. However, I've changed it so that we set bindings (to be empty) here, rather than setting bindings_err.